README.md

```diff
@@ -262,7 +264,158 @@ export default NuxtAuthHandler({
+```ts
+// Strapi v4 url, out of the box
+ STRAPI_BASE_URL=http://localhost:1337/api
+ STRAPI_API_TOKEN_SECRET=optional
+```
+
+In your `nuxt.config.ts` file, configure it like the following
+```ts
+export default defineNuxtConfig({
+  runtimeConfig: {
+    // The private keys which are only available server-side
+    NUXT_SECRET: process.env.NUXT_SECRET,
+    // Default http://localhost:1337/api
+    STRAPI_BASE_URL: process.env.STRAPI_BASE_URL,
+    // Optional but required for server-to-server ops
+    STRAPI_APITOKEN_SECRET:process.env.STRAPI_APITOKEN_SECRET
+  },
+  auth: {
+    origin: process.env.ORIGIN,
+  },
+});
+
+```
+
+Create the catch-all nitro route and add the this custom Strapi credentials provider.
+```ts
+// file: ~/server/api/auth/[...].ts
+import CredentialsProvider from "next-auth/providers/credentials";
+import { NuxtAuthHandler } from "#auth";
+
+export default NuxtAuthHandler({
+  // secret needed to run nuxt-auth in production mode (used to encrypt data)
+  secret: process.env.NUXT_SECRET,
+  providers: [
+    // @ts-ignore Import is exported on .default during SSR, so we need to call it this way. May be fixed via Vite at some point
+    CredentialsProvider.default({
+      name: "Credentials",
+      credentials: {
+        username: { label: "Username", type: "text", placeholder: "Test user" },
+        password: { label: "Password", type: "password" },
+      },
+      async authorize(credentials: any) {
+        
+        const user = await $fetch(
+          `${process.env.STRAPI_BASE_URL}/auth/local/`,
+          {
+            method: "POST",
+            body: JSON.stringify({
+              identifier: credentials.username,
+              password: credentials.password,
+            }),
+          }
+        );
+
+        if (user) {
+          const u = {
+            id: user.user.id,
+            name: user.user.username,
+            email: user.user.email,
+            // Passing OG JWT through the email field.
+            // IMPORTANT: Always pass around encoded JWTs,
+            // don't pass around decoded JWTs.
+            profile: user.jwt
+          };
+          return u;
+        } else {
+          return null;
+        }
+      },
+    }),
+  ],
+  session: {
+    jwt: true,
+  }
+});
+```
+
+###### Passing the headers after authenticating. 
+
+After setting up the authentication flow and when authenticated in the browser.
+We need to supply the cookie headers to Nitro in our fetch calls to access auth protected Strapi resources.
+- Create a useFetch or $fetch request that calls Nitro that includes the session cookie headers.
+- In order to gain access to your protected Strapi endpoints, we need to extract the user object from the session server side in nitro using the getToken function.
+- Finally we add the JWT token stored in the session user object as auth headers for our fetch request to Strapi.
+
+Add the following fetch request in e.g. your admin dashboard section of your Nuxt app.
```

Comment:
this is the first time you mention this section, so I and others will not know what it is (:

Can you rework these "usage" sections to build up from zero? E.g.:
1. how to do a  app-side call to strapi by passing the strapi JWT token as a `Authorization` header
2. create an endpoint in the nuxt app that does the data fetching from a protected strapi endpoint, then call that endpoint from the frontend

Reply:
I added an example, but it went out of scope according to a newer change request.